backend/regalloc/regalloc_invariants.ml

```diff
@@ -146,7 +147,7 @@ let postcondition_layout : Cfg_with_layout.t -> unit =
       unit =
     match reg.Reg.loc with
     | Reg phys_reg ->
-      let phys_reg = Proc.phys_reg phys_reg in
+      let phys_reg = Proc.phys_reg reg.typ phys_reg in
```

Comment:
I think passing the type guarantees
the following condition cannot be
true.

Reply:
Good point, changed to check the stack slot class instead

Reply:
Yeah, if we merge int/float this would be insufficient. Fixed

Reply:
I am still not convinced: we used to
check that `reg` and `phys_reg`, taken
independently, were compatible.
We now check they are compatible,
but information from `reg` is used to
build `phys_reg`.

So I am unsure we are really checking
anything. And the condition is probably
redundant: if registers are in the same
class, aren't they necessarily in the same
stack class?

Perhaps the check can no longer exist,
given the new `Proc`.

Reply:
Updated to just run `phys_reg`